src/client.rs

```diff
@@ -1091,14 +1093,13 @@ impl<S, T> Drop for Client<S, T> {
 
         // Dirty shutdown
         // TODO: refactor, this is not the best way to handle state management.
-        if let Some(address_id) = self.last_address_id {
-            self.stats.client_disconnecting(self.process_id, address_id);
-
-            if self.connected_to_server {
-                if let Some(process_id) = self.last_server_id {
-                    self.stats.server_idle(process_id, address_id);
-                }
-            }
+        self.stats.client_disconnecting(self.process_id, 0);
+        if self.connected_to_server
+            && self.last_server_id.is_some()
+            && self.last_address_id.is_some()
+        {
+            self.stats
+                .server_idle(self.last_server_id.unwrap(), self.last_address_id.unwrap());
```

Comment:
We register any clients that connects, even if it does not perform queries. As such, we must also deregister any client that disconnects.

Reply:
We're going back and forth on this one I think, didn't we remove this call at some point?

Reply:
I think this was `server_idle`, I don't recall we touched `client_disconnecting`.

Reply:
I feel a bit more confident about that stats after the refactor, having tracking objects for individual connections + registering all connections as early as possible makes it easier to reason about.